### PR TITLE
PR: Fix traitlets deprecation warning

### LINKED ISSUE #14534: DeprecationWarning: ShellWidget._syntax_style_changed is deprecated in traitlets 4.1: use @observe and @unobserve instead
I received the following warning when running the test suite:

```
spyder/plugins/ipythonconsole/tests/test_ipythonconsole.py::test_banners
  <<<PACKAGEDIR>>>/spyder/plugins/ipythonconsole/widgets/shell.py:789: DeprecationWarning: ShellWidget._syntax_style_changed is deprecated in traitlets 4.1: use @observe and @unobserve instead.
    def _syntax_style_changed(self):

```

### PR DESCRIPTION
**Pull request by juliangilbey**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

Fixed the traitlets DeprecationWarning raised by `spyder/plugins/ipythonconsole/tests/test_ipythonconsole.py::test_banners`.  The unit test should no longer raise a warning after this change.

### Issue(s) Resolved

Fixes #14534 with thanks to @dalthviz 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org),
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @juliangilbey

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -241,6 +241,7 @@ def run(self):
     'spyder-kernels>=2.2.0,<2.3.0',
     'textdistance>=4.2.0',
     'three-merge>=0.1.1',
+    'traitlets>=4.1',
```
Comment (ccordoba12) [inline setup.py:244]:
```suggestion
```

`trailets` is a transitive dependency, so we don't need to depend on it directly:

    spyder -> spyder-kernels -> ipykernel -> traitlets

**Review by ccordoba12 — COMMENTED** [review]
Thanks for your help with this @juliangilbey! I left a minor comment, otherwise looks good to me.

File: setup.py (hunk 1 above)
Comment (juliangilbey) [reply setup.py:244]: This is true, though the remainder of the patch does explicitly import it.  I guess it's a matter of style to a great extent; this explicit listing just makes the version requirements explicit, though ipykernel does that anyway.  I'm not particularly bothered either way; happy to make that change or let you do it.

Comment (ccordoba12) [reply setup.py:244]: Please make it. We already have a lot of dependencies to add one more for such a simple change.

Comment (juliangilbey) [reply setup.py]: Done!

**Review by ccordoba12 — APPROVED** [review]
Thanks @juliangilbey for your help with this!

**Pull request merged**